lib/spack/spack/util/module_cmd.py

```diff
@@ -32,8 +32,7 @@ def module(*args):
     if args[0] in module_change_commands:
         # Do the module manipulation, then output the environment in JSON
         # and read the JSON back in the parent process to update os.environ
-        module_cmd += ' > /dev/null; PYTHONHOME="{0}" "{1}" -c "{2}"'.format(
-            sys.prefix, sys.executable, py_cmd)
+        module_cmd += ' >/dev/null;' + sys.executable + ' -c %s' % py_cmd
```

Comment:
@adamjstewart I'm not sure this needs changing, as probably it's always called before the modifications from packages are applied. Need to double check though.